crates/swc_icu_messageformat_parser/src/parser.rs

```diff
@@ -831,8 +831,14 @@ impl<'s> Parser<'s> {
         while !self.is_eof() && is_potential_element_name_char(self.char()) {
             self.bump();
         }
-
-        &self.message[start_offset..self.offset()]
+        #[cfg(feature = "utf16")]
```

Comment:
I think this should not use `Box::leak`

Reply:
I'm not sure how to do this: slice a`Utf16String` to `&str` and return

Reply:
I think without a leak I cannot borrow from a new utf8string.

Reply:
Maybe I should find the correct position from message?